src/pages/zoeken/[slug]/[ie]/[name]/index.tsx

```diff
@@ -502,6 +525,16 @@ const ObjectDetailPage: NextPage<ObjectDetailPageProps> = ({ title, description,
 	};
 
 	const onRequestMaterialClick = () => {
+		if (
+			materialRequests?.items &&
+			materialRequests.items.find(
+				(request) => request.objectSchemaIdentifier === mediaInfo?.schemaIdentifier
+			)
+		) {
+			onDuplicateRequest();
+			return;
+		}
+
```

Comment:
it might be easier to just fetch the material requests when a user clicks on this button and check if it is in the list
then we don't need all the above logic to keep the list in sync

We usually want to use the react query wrappers so we get loading states, but in this case, we want to check something when the user clicks a button, so an await call directly to the service is simpler

```suggestion
		const materialRequests = await MaterialRequestsService.getAll({
			size: 500,
			isPending: true,
			isPersonal: true,
		});

		if (
			materialRequests?.items &&
			materialRequests.items.find(
				(request) => request.objectSchemaIdentifier === mediaInfo?.schemaIdentifier
			)
		) {
			onDuplicateRequest();
			return;
		}

```